gm4_zauber_liquids/data/gm4_zauber_liquids/functions/item_drain/zauber_wormhole_potion.mcfunction

```diff
@@ -0,0 +1,27 @@
+# Non-Weighted Averages: (oldcoord+newcoord)/2
+execute store result score math gm4_zl_warp_cx run data get block ~ ~ ~ Items[{Slot: 0b}].tag.gm4_zauber_cauldrons.cauldron_pos.x
+execute store result score math gm4_zl_warp_cy run data get block ~ ~ ~ Items[{Slot: 0b}].tag.gm4_zauber_cauldrons.cauldron_pos.y
+execute store result score math gm4_zl_warp_cz run data get block ~ ~ ~ Items[{Slot: 0b}].tag.gm4_zauber_cauldrons.cauldron_pos.z
+execute store result score math gm4_zl_warp_cd run data get block ~ ~ ~ Items[{Slot: 0b}].tag.gm4_zauber_cauldrons.cauldron_pos.dimension
```

Comment:
Similar to the filling, draining may be automated, therefore efficiency is of essence. Pulling the entire NBT into a storage before extracting the scores might be beneficial.
```suggestion
# Non-Weighted Averages: (oldcoord+newcoord)/2
data modify storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item.tag.gm4_zauber_cauldrons.cauldron_pos set from block ~ ~ ~ Items[{Slot: 0b}].tag.gm4_zauber_cauldrons.cauldron_pos
execute store result score math gm4_zl_warp_cx run data get storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item.tag.gm4_zauber_cauldrons.cauldron_pos.x
execute store result score math gm4_zl_warp_cy run data get storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item.tag.gm4_zauber_cauldrons.cauldron_pos.y
execute store result score math gm4_zl_warp_cz run data get storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item.tag.gm4_zauber_cauldrons.cauldron_pos.z
execute store result score math gm4_zl_warp_cd run data get storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item.tag.gm4_zauber_cauldrons.cauldron_pos.dimension
data remove storage gm4_zauber_liquids:temp/item/drain/zauber_wormhole Item
```